fe1-web/src/features/linked-organizations/components/AddLinkedOrganizationModal.tsx

```diff
@@ -0,0 +1,300 @@
+import { dispatch } from 'core/redux';
+import { Spacing, Typography, Color, ModalStyles } from 'core/styles';
+import { FOUR_SECONDS } from 'resources/const';
+import STRINGS from 'resources/strings';
+
+import { LinkedOrganizationsHooks } from '../hooks';
+import { expectFederation, initFederation, requestChallenge } from '../network';
+import { Challenge } from '../objects/Challenge';
+import { LinkedOrganization } from '../objects/LinkedOrganization';
+import { makeChallengeSelector } from '../reducer';
+import { addLinkedOrganization } from '../reducer/LinkedOrganizationsReducer';
+import ManualInputModal from './ManualInputModal';
+import QRCodeModal from './QRCodeModal';
+import QRCodeScannerModal from './QRCodeScannerModal';
+
+type NavigationProps = CompositeScreenProps<
+  StackScreenProps<
+    LinkedOrganizationsParamList,
+    typeof STRINGS.linked_organizations_navigation_addlinkedorgModal
+  >,
+  CompositeScreenProps<
+    StackScreenProps<LaoParamList, typeof STRINGS.navigation_lao_linked_organizations>,
+    StackScreenProps<AppParamList, typeof STRINGS.navigation_app_lao>
+  >
+>;
+
+const styles = StyleSheet.create({
+  modalContainer: {
+    ...ModalStyles.modalContainer,
+  },
+  marginBottomSpacingX1: {
+    marginBottom: Spacing.x1,
+  },
+  infoText: {
+    textAlign: 'center',
+    fontSize: 18,
+    margin: 3,
+    color: Color.contrast,
+  },
+});
+
+const AddLinkedOrganizationModal = () => {
+  const navigation = useNavigation<NavigationProps['navigation']>();
+  const toast = useToast();
+  const laoId = LinkedOrganizationsHooks.useCurrentLaoId();
+  const lao = LinkedOrganizationsHooks.useCurrentLao();
+  const challengeSelector = useMemo(() => makeChallengeSelector(laoId), [laoId]);
+  const challengeState = useSelector(challengeSelector);
+
+  const [showModal, setShowModal] = useState<boolean>(true);
+
+  const [showQRScannerModal, setShowQRScannerModal] = useState<boolean>(false);
+  const [showQRCodeModal, setShowQRCodeModal] = useState<boolean>(false);
+
+  const [inputModalIsVisible, setInputModalIsVisible] = useState(false);
+
+  const [qrCodeData, setQRCodeData] = useState<string>('');
+  // Check if the organizer is the one thats supposed to send the initFederation message
+  const [isInitiatingOrganizer, setIsInitiatingOrganizer] = useState<boolean>(false);
+  // this is needed as otherwise the camera may stay turned on
+  const [showScanner, setShowScanner] = useState(false);
+
+  const onRequestChallenge = useCallback(() => {
+    requestChallenge(laoId)
+      .then(() => {})
+      .catch((err) => {
+        console.error('Could not request Challenge, error:', err);
+        toast.show(`Could not request Challenge, error: ${err}`, {
+          type: 'danger',
+          placement: 'bottom',
+          duration: FOUR_SECONDS,
+        });
+      });
+  }, [laoId, toast]);
+
+  const getQRCodeData = () => {
```

Comment:
In general I would try to avoid prefixing functions with side effects (and not even returning a value) with `get` :) Is there a better name you can think of that may explain what the function actually does? (It definitely does not just get some value and return it, i.e. what you usually expect from a getX function)